homework/index.js

```diff
@@ -30,17 +32,122 @@
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  // a function to create td without link
+  function createTd(textKey, textContent, tbody) {
+    const parent = createAndAppend('tr', tbody);
+    createAndAppend('td', parent, {
+      text: `${textKey}: `,
+      class: 'label',
+    });
+    createAndAppend('td', parent, {
+      text: textContent,
+    });
+  }
+  // table of repository's basic information
+  function createDetailsTable(repository, container) {
+    const repositoryData = createAndAppend('div', container, { class: 'left-div white-frame' });
+    const table = createAndAppend('table', repositoryData);
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, {
+      text: 'Repository :',
+      class: 'label',
+    });
+    // manually create td with a link
+    const td = createAndAppend('td', tr);
+    createAndAppend('a', td, {
+      href: repository.html_url,
+      text: repository.name,
+      target: '_blank',
+    });
+    createTd('Description', repository.description, tbody);
+    createTd('Forks', repository.forks, tbody);
+    createTd('Updated', repository.updated_at, tbody);
+  }
+
+  // contributors list
+  function listContributors(repository, container) {
+    const contributorUrl = repository.contributors_url;
+    fetchJSON(contributorUrl)
+      .then(contributors => {
+        const contributorsData = createAndAppend('div', container, {
+          class: 'right-div white-frame',
+        });
+        createAndAppend('p', contributorsData, {
+          text: 'contributions',
+          class: 'contributor-header',
+        });
+        const ul = createAndAppend('ul', contributorsData, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul, {
+            class: 'contributor-item',
+          });
+          // The open() method creates a new secondary browser window,
+          // a new blank, empty window (URL about:blank) is created with the default toolbars of the main window.
+          li.addEventListener('click', () => {
+            window.open(contributor.html_url);
+          });
+
+          createAndAppend('img', li, {
+            src: contributor.avatar_url,
+            alt: "contributor's profile",
+            class: 'contributor-avatar',
+          });
+          const liDiv = createAndAppend('div', li, { class: 'contributor-data' });
+          createAndAppend('div', liDiv, { text: contributor.login });
+          createAndAppend('div', liDiv, {
+            text: contributor.contributions,
+            class: 'contributor-badge',
+          });
+        });
```

Comment:
One of the requirements for ARIA compliance is that you must be able to navigate the UI with the keyboard. At present you cannot go to a contributor list item with the keyboard and activate it to go to the contributor's GitHub page by pressing Enter. The simplest way to achieve that is to wrap the `<li>` content into an `<a>` tag like this:

```js
contributors.forEach(contributor => {
  const li = createAndAppend('li', ul);
  const anchor = createAndAppend('a', li, {
    href: contributor.html_url,
    target: '_blank',
    class: 'contributor-item',
  });
  createAndAppend('img', anchor, {
    src: contributor.avatar_url,
    alt: "contributor's profile",
    class: 'contributor-avatar',
  });
  const liDiv = createAndAppend('div', anchor, { class: 'contributor-data' });
  createAndAppend('div', liDiv, { text: contributor.login });
  createAndAppend('div', liDiv, {
    text: contributor.contributions,
    class: 'contributor-badge',
  });
});
```

You will need to tweak the `contributor-item` CSS class a bit to get rid of the underline decoration that you normally get for an `<a>` tag.

Reply:
Sure, I’ll do that.
Thanks for your helpful feedback!

Reply:
May I ask you a Q?
Shall I use this way way in place of the `eventListner` ?
```li.addEventListener('click', () => {
            window.open(contributor.html_url);
          });```

Reply:
You don't need / shouldn't use an event listener at all if you use an `<a>` tag. If your click and `<a>` tag it will automatically go to the `url` specified by its `href` attribute. That is the whole purpose of an `<a>` tag. It is also by default included in the tab order (you can uset the tab key to navigate to an `<a>` tag) and also activates the link if you press Enter while the `<a>` tag has the focus.

Reply:
Good to know!
Thanks for your response!